src/NovelRT.Interop/ResourceManagement/NrtResourceLoader.cpp

```diff
@@ -0,0 +1,76 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root
+// for more information.
+
+#include "../LifetimeExtender.h"
+#include <NovelRT.Interop/NrtErrorHandling.h>
+#include <NovelRT.Interop/ResourceManagement/NrtResourceLoader.h>
+#include <NovelRT/ResourceManagement/ResourceManagement.h>
+
+using namespace NovelRT::ResourceManagement;
+using namespace NovelRT::Interop;
+
+NrtResult Nrt_ResourceLoader_GetResourcePath(NrtResourceLoaderHandle loader, char** outputPath)
+{
+    if (loader == nullptr)
+    {
+        Nrt_setErrIsNullInstanceProvidedInternal();
+        return NRT_FAILURE_NULL_INSTANCE_PROVIDED;
+    }
+
+    if (outputPath == nullptr)
+    {
+        Nrt_setErrIsNullArgProvidedInternal();
+        return NRT_FAILURE_NULL_ARGUMENT_PROVIDED;
+    }
+
+    auto& path = reinterpret_cast<ResourceLoader*>(loader)->ResourcesRootDirectory();
+    auto pathString = path.string();
+
+#ifdef WIN32
+    size_t length = pathString.length() + 1;
+    *outputPath = static_cast<char*>(malloc(length));
+    strcpy_s(*outputPath, length, pathString.c_str());
+#else
+    *outputPath = strdup(pathString.c_str());
+#endif
```

Comment:
We are no longer using the _s variants. Please put in the explicit checks required to use the function that exists in all places for what you require.

Reply:
Wouldn't it be better to create a function in `NrtMisc.h` containing this piece of code (with strcpy) for copying strings?

Reply:
Not sure when we decided this, @RubyNova ? It's still used in two other places and I couldn't find history of it - please open an issue to cover the spots where this is still used if that's the case?

Reply:
We already have an open issue, see #388

Reply:
Ah looks like someone's been assigned it already - cool beans